pkg/controllers/provisioning/scheduling/inflightnode.go

```diff
@@ -87,9 +90,18 @@ func (n *InFlightNode) Add(pod *v1.Pod) error {
 		return err
 	}
 
-	if err := n.hostPortUsage.Add(pod); err != nil {
+	if err := n.hostPortUsage.Validate(pod); err != nil {
+		return err
+	}
+
+	// determine the number of volumes that will be mounted if the pod schedules
+	mountedVolumeCount, err := n.volumeUsage.Validate(pod)
+	if err != nil {
 		return err
 	}
+	if mountedVolumeCount.Exceeds(n.maxVolumes) {
+		return fmt.Errorf("would exceed max volume")
```

Comment:
Does this print something useful to the user when unwrapped all the way?
```suggestion
		return fmt.Errorf("would exceed node volume limits")
```